lib/puppet/provider/iis_site/webadministration.rb

```diff
@@ -0,0 +1,182 @@
+require_relative '../../../puppet_x/puppetlabs/iis/powershell_manager'
+
+# When writing IIS PowerShell code for any of the methods below
+# NEVER EVER use Get-Website without specifying -Name. As the number
+# of sites on a server increases, Get-Website will take longer and longer
+# to return. This will exponentially increase the total duration of your
+# puppet run
+
+Puppet::Type.type(:iis_site).provide(:webadministration) do
+  desc "IIS Provider using the PowerShell WebAdministration module"
+
+  confine    :operatingsystem => [:windows ]
+  defaultfor :operatingsystem => :windows
+
+  commands :powershell =>
+    if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
+    elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
+    else
+      'powershell.exe'
+    end
+
+  mk_resource_methods
+
+  def create
+    cmd = []
+
+    cmd << self.class.ps_script_content('_newwebsite', @resource)
+
+    cmd << self.class.ps_script_content('generalproperties', @resource)
+
+    cmd << self.class.ps_script_content('logproperties', @resource)
+
+    cmd << self.class.ps_script_content('serviceautostartprovider', @resource)
+
+    inst_cmd = cmd.join
+
+    result = self.class.run(inst_cmd)
+
+    if result[:exitcode] != 0
+      fail("Error creating website: #{result[:errormessage]}")
+      return false
```

Comment:
I don't think we'll ever get to this `return` as `fail` will raise an exception and we'll return context to the caller.